unit/rates.go

```diff
@@ -0,0 +1,289 @@
+func (s SatPerKWeight) FeeForWeight(wu WeightUnit) btcutil.Amount {
+	// The resulting fee is rounded down, as specified in BOLT#03.
+	fee := new(big.Rat).Mul(
+		s.Rat, big.NewRat(safeUint64ToInt64(uint64(wu)), SatsPerKilo),
+	)
+
+	return btcutil.Amount(new(big.Int).Div(fee.Num(), fee.Denom()).Int64())
+}
+
+// FeeForWeightRoundUp calculates the fee resulting from this fee rate and the
+// given weight in weight units (wu), rounding up to the nearest satoshi.
+func (s SatPerKWeight) FeeForWeightRoundUp(
+	wu WeightUnit) btcutil.Amount {
+	// The rounding logic is based on the ceiling division formula:
+	// (numerator + denominator - 1) / denominator
+	//
+	// This ensures that any fractional part of the fee is rounded up to
+	// the next whole satoshi.
+	feeRat := new(big.Rat).Mul(
+		s.Rat, big.NewRat(safeUint64ToInt64(uint64(wu)), SatsPerKilo),
+	)
+
+	num := feeRat.Num()
+	den := feeRat.Denom()
+	num.Add(num, den)
+	num.Sub(num, big.NewInt(1))
+	num.Div(num, den)
+
+	return btcutil.Amount(num.Int64())
+}
+
+// FeeForVByte calculates the fee resulting from this fee rate and the given
+// size in vbytes (vb).
+func (s SatPerKWeight) FeeForVByte(vb VByte) btcutil.Amount {
+	return s.FeePerKVByte().FeeForVSize(vb)
+}
+
+// FeePerKVByte converts the current fee rate from sat/kw to sat/kb.
+func (s SatPerKWeight) FeePerKVByte() SatPerKVByte {
+	kwToKvbRate := big.NewRat(blockchain.WitnessScaleFactor, 1)
+	kvbRate := new(big.Rat).Mul(s.Rat, kwToKvbRate)
+
+	return SatPerKVByte{kvbRate}
+}
+
+// FeePerVByte converts the current fee rate from sat/kw to sat/vb.
+func (s SatPerKWeight) FeePerVByte() SatPerVByte {
+	kwToVbRate := big.NewRat(blockchain.WitnessScaleFactor, SatsPerKilo)
+	vbRate := new(big.Rat).Mul(s.Rat, kwToVbRate)
+
+	return SatPerVByte{vbRate}
+}
+
+// String returns a human-readable string of the fee rate.
+func (s SatPerKWeight) String() string {
+	return s.FloatString(floatStringPrecision) + " sat/kw"
+}
+
+// Equal returns true if the fee rate is equal to the other fee rate.
+func (s SatPerKWeight) Equal(other SatPerKWeight) bool {
+	return s.Cmp(other.Rat) == 0
+}
+
+// GreaterThan returns true if the fee rate is greater than the other fee rate.
+func (s SatPerKWeight) GreaterThan(other SatPerKWeight) bool {
+	return s.Cmp(other.Rat) > 0
+}
+
+// LessThan returns true if the fee rate is less than the other fee rate.
+func (s SatPerKWeight) LessThan(other SatPerKWeight) bool {
+	return s.Cmp(other.Rat) < 0
+}
+
+// GreaterThanOrEqual returns true if the fee rate is greater than or equal to
+// the other fee rate.
+func (s SatPerKWeight) GreaterThanOrEqual(other SatPerKWeight) bool {
+	return s.Cmp(other.Rat) >= 0
+}
+
+// LessThanOrEqual returns true if the fee rate is less than or equal to the
+// other fee rate.
+func (s SatPerKWeight) LessThanOrEqual(other SatPerKWeight) bool {
+	return s.Cmp(other.Rat) <= 0
+}
+
+// round rounds a big.Rat to the nearest btcutil.Amount (int64).
+
+func round(r *big.Rat) btcutil.Amount {
```

Comment:
```suggestion
// roundToAmount rounds a big.Rat to the nearest btcutil.Amount (int64).
func roundToAmount(r *big.Rat) btcutil.Amount {
```

Not a blocker, but I think renaming the function to `roundToAmount` makes it clearer when reading the code, since it specifies the target type. Also, there’s a blank line below the comment.

Reply:
done